lib/puppet/type/cs_location.rb

```diff
@@ -44,6 +45,24 @@ module Puppet
         defaultto 'INFINITY'
     end
 
+    newproperty(:boolean) do
```

Comment:
This allows users to define whether the rule is AND/OR. See boolean-op below (taken from the cib)...

      <rsc_location id="location-ha_vip" rsc="ha_vip">
        <rule boolean-op="or" id="location-ha_vip-rule" score="500">
          <expression attribute="ethmonitor" id="location-ha_vip-rule-expr-1" operation="lt" value="1"/>
          <expression attribute="ethmonitor" id="location-ha_vip-rule-expr-2" operation="not_defined"/>
        </rule>
      </rsc_location>